BlazarUI/app/scripts/components/sidebar/SidebarItem.jsx

```diff
@@ -1,92 +1,161 @@
 import React, {Component, PropTypes} from 'react';
+import {Link} from 'react-router'
 import classnames from 'classnames';
 import {has, contains} from 'underscore';
 import {truncate} from '../Helpers.js';
 import BuildingIcon from '../shared/BuildingIcon.jsx';
 import Icon from '../shared/Icon.jsx';
 import Star from '../shared/Star.jsx';
+import BuildStates from '../../constants/BuildStates.js';
 import {buildResultIcon} from '../Helpers.js';
 
-import {Link} from 'react-router'
+let initialState = {
+  expanded: false
+};
 
 class SidebarItem extends Component {
 
+  constructor() {
+    this.state = initialState;
+  }
+
   getItemClasses() {
     return classnames([
       'sidebar-item',
       this.props.classNames
     ]);
   }
+
+  toggleExpand() {
+    this.setState({
+      expanded: !this.state.expanded
+    });
+  }
+
+  renderExpandText(numberRemaining) {
+    const {builds} = this.props;
+
+    if (builds.length < 4) {
+      return '';
+    }
+
+    else if (!this.state.expanded) {
+      return (
+        <div onClick={this.toggleExpand.bind(this)} className='sidebar-item__and-more'>
+          show {numberRemaining} more
+        </div>
+      );
+    }
+
+    const numberToHide = builds.length - 3;
+
+    return (
+      <div onClick={this.toggleExpand.bind(this)} className='sidebar-item__and-more'>
+        show {numberToHide} less
```

Comment:
For sure! Going to avoid the one liner and the cleverness of `show` separated from the rest of message in favor of the slightly more readable and verbose series of `if`s